spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+RSpec.describe Post, type: :feature do
+  describe 'index page' do
+    before(:each) do
+      @user = User.create(
+        name: 'Tascha',
+        photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+        bio: 'I love singing',
+        post_counter: 0
+      )
+      @post = Post.create(
+        title: 'Programming school at Microverse is awesome',
+        text: 'Coding is fun',
+        comments_counter: 0,
+        likes_counter: 1,
+        author_id: @user.id
+      )
+      Comment.create(author_id: @user, post: @post, text: 'Lets do this')
+      @users = User.all
+      visit user_posts_path(@users.ids)
+    end
+    it "should show the user's username." do
+      visit user_posts_path(@user.id)
+      expect(page).to have_content('Tascha')
+    end
+    it "should show the user's profile picture." do
+      visit user_posts_path(@user.id)
+      expect(page).to have_xpath("//img[contains(@src,'https://unsplash.com/photos/F_-0BxGuVvo')]")
+    end
+    it 'should show the number of posts each user has written.' do
+      visit user_posts_path(@user.id)
+      expect(page).to have_content('Coding is fun')
+    end
+    it "should show a post's title." do
+      expect(page).to have_content('Programming school at Microverse is awesome')
+    end
+    it "I can see some of the post's body." do
+      expect(page).to have_content('Programming school at Microverse is awesome')
+    end
+    it 'should show the first comments on a post' do
+      expect(page).to have_content(@post.comments.first)
+    end
+    it 'should show how many comments a post has.' do
+      expect(page).to have_content(@post.comments_counter)
+    end
+    it 'should show how many likes a post has.' do
+      expect(page).to have_content(@post.likes_counter)
+    end
+  end
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20a%20section%20for%20pagination%20if%20there%20are%20more%20posts%20than%20fit%20on%20the%20view.), Please add a test that asserts you can see a section for pagination if there are more posts than fit on the view.
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=When%20I%20click%20on%20a%20post%2C%20it%20redirects%20me%20to%20that%20post%27s%20show%20page.). Please add a Test that redirects to the post's show page when you click on a post.